packages/gatsby/src/utils/webpack-status.ts

```diff
@@ -0,0 +1,18 @@
+import { flush, isFlushEnqueued } from "./page-data"
+
+let isPendingStatus = false
+
+export function isPending(): boolean {
+  return isPendingStatus
+}
+
+export function markAsPending(): void {
+  isPendingStatus = true
+}
+
+export function markAsDone(): void {
+  isPendingStatus = false
+  if (isFlushEnqueued()) {
```

Comment:
you could make this just a `flush` call and have it exit early if there isn't any flush enqueued

Reply:
But then it needs to set the `isPendingStatus` flag to false as well 🤷‍♂️